tests/Unit/DataStructures/Test_Mesh.cpp

```diff
@@ -195,3 +222,42 @@ SPECTRE_TEST_CASE("Unit.Serialization.Mesh",
               {{Spectral::Quadrature::GaussLobatto, Spectral::Quadrature::Gauss,
                 Spectral::Quadrature::GaussLobatto}}});
 }
+
+// [[OutputRegex, Tried to slice through non-existing dimension]]
+[[noreturn]] SPECTRE_TEST_CASE(
+    "Unit.DataStructures.Mesh.SliceThroughNonExistingDimension",
+    "[DataStructures][Unit]") {
+  ASSERTION_TEST();
+#ifdef SPECTRE_DEBUG
+  const Mesh<1> mesh1d{2, Spectral::Basis::Legendre,
+                       Spectral::Quadrature::GaussLobatto};
+  mesh1d.slice_through(1);
+  ERROR("Failed to trigger ASSERT in an assertion test");
+#endif
+}
+
+// [[OutputRegex, Tried to slice away non-existing dimension]]
+[[noreturn]] SPECTRE_TEST_CASE(
+    "Unit.DataStructures.Mesh.SliceAwayNonExistingDimension",
+    "[DataStructures][Unit]") {
+  ASSERTION_TEST();
+#ifdef SPECTRE_DEBUG
+  const Mesh<1> mesh1d{2, Spectral::Basis::Legendre,
+                       Spectral::Quadrature::GaussLobatto};
+  mesh1d.slice_away(1);
+  ERROR("Failed to trigger ASSERT in an assertion test");
+#endif
+}
+
+// [[OutputRegex, Dimensions to slice through contain duplicates]]
+[[noreturn]] SPECTRE_TEST_CASE(
+    "Unit.DataStructures.Mesh.SliceThroughDuplicateDimensions",
+    "[DataStructures][Unit]") {
+  ASSERTION_TEST();
+#ifdef SPECTRE_DEBUG
+  const Mesh<2> mesh2d{2, Spectral::Basis::Legendre,
+                       Spectral::Quadrature::GaussLobatto};
+  mesh2d.slice_through(0, 0);
```

Comment:
It would be useful to test the case we were discussing, `(2, 1,1 )` since this requires the most complicated logic

Reply:
Fixed